### PR TITLE
Streaming Feedback

### PR DESCRIPTION
**Pull request by jburns20**
The front-end for streaming feedback.

### CODE REVIEW

File: app/views/assessments/_downloadSubmission.html.erb [hunk 1]
```diff
@@ -20,16 +20,17 @@
...
+          feedback: problem.id}, title: "In-progress autograder output" %>
+    <% elsif @scores[sub.id] and @scores[sub.id][problem.id] and @scores[sub.id][problem.id][:released] == 1 then
       score = @scores[sub.id][problem.id]
       if score[:feedback] then %>
         <%= link_to ("output"), { action: :viewFeedback,
           submission_id: sub.id,
           feedback: problem.id}, title: "Autograding output file" %>
       <% end %>
-    <% else %>
-      <%= link_to "status", [@course, :jobs], title: "View a history of autograding jobs" %>
```
Comment (ncik-roberts) [inline app/views/assessments/_downloadSubmission.html.erb:32]: Why remove this?

File: db/schema.rb [hunk 2]
```diff
@@ -97,6 +97,7 @@
     t.text     "embedded_quiz_form_data", limit: 65535
     t.boolean  "embedded_quiz",           limit: 1
     t.binary   "embedded_quiz_form"
+    t.boolean  "grade_latest",            limit: 1,     default: false
```
Comment (ncik-roberts) [inline db/schema.rb]: This seems to be from the branch where I'm working on grading the max-scoring submission. It doesn't have an accompanying schema patch here, so we should probably remove it.

File: app/controllers/assessments_controller.rb [hunk 3]
```diff
@@ -527,6 +527,16 @@ def viewFeedback
...
+    output_file = "#{@submission.course_user_datum.email}_#{@submission.version}_#{@assessment.name}_autograde.txt"
+    response = TangoClient.poll("#{@course.name}-#{@assessment.name}", "#{URI.encode(output_file)}", in_progress = 1)
+    # json is returned when a job is not complete
+    if response.content_type == "application/json"
+      @feedback = "Output file not found. Please try again later."
+    else
+      @feedback = response.body
+    end
+  rescue TangoClient::TangoException => e
+    @feedback = "Output file not found. Please try again later."
```
Comment (ncik-roberts) [inline app/controllers/assessments_controller.rb]: One time this error can happen is if the job hasn't started running yet. We should probably say as much in the error message.

**Review by ncik-roberts — APPROVED** [review]

File: app/views/assessments/_downloadSubmission.html.erb (hunk 1 above)
Comment (jburns20) [reply app/views/assessments/_downloadSubmission.html.erb:32]: It's just a link to the jobs page, which is already in the header.

File: db/schema.rb (hunk 2 above)
Comment (jburns20) [reply db/schema.rb]: oops, yeah.

**Pull request merged**